workspaces/arborist/lib/arborist/reify.js

```diff
@@ -717,8 +717,15 @@ module.exports = cls => class Reifier extends cls {
     // ${REGISTRY} or something.  This has to be threaded through the
     // Shrinkwrap and Node classes carefully, so for now, just treat
     // the default reg as the magical animal that it has been.
-    return resolved && resolved
-      .replace(/^https?:\/\/registry\.npmjs\.org\//, this.registry)
+    const resolvedURL = new URL(resolved)
```

Comment:
if `resolved` is undefined, an empty string, or any other falsey value this constructor will throw an error. i think this needs to be moved inside the `if (resolved) {}` condition

Reply:
This is a private method that is only called once resolved is confirmed not falsey. At @nlf's suggestion, removed the condition and added a comment.